projects/plugins/boost/app/assets/src/js/utils/analytics.ts

```diff
@@ -73,3 +73,13 @@ function addBoostProps( props: TracksEventProperties ): TracksEventProperties {
 
 	return { ...defaultProps, ...props };
 }
+
+export function recordBoostEventAndRedirect(
+	url: string,
+	eventName: string,
+	eventProp: TracksEventProperties
+) {
+	recordBoostEvent( eventName, eventProp ).then( () => {
+		window.location.href = url;
+	} );
+}
```

Comment:
I think this would be easier to read as an `async` function. 

```suggestion
export async function recordBoostEventAndRedirect(
	url: string,
	eventName: string,
	eventProp: TracksEventProperties
) {
	await recordBoostEvent( eventName, eventProp );
	window.location.href = url;
}
```

This also has the added benefit that if `recordBoostEvent` rejects/throws, then so will this function. So callers can know for sure that it will either do what it says on the tin, or will throw an error. :)